synapse/lib/cell.py

```diff
@@ -753,17 +757,79 @@ async def runBackup(self, name=None, wait=True):
             mesg = 'Backup with name already exists'
             raise s_exc.BadArg(mesg=mesg)
 
-        task = self.schedCoro(self._execBackTask(path))
+        task = self.schedCoro(self._execBackupTask(path))
 
         if wait:
             await task
 
         return name
 
-    async def _execBackTask(self, dirn):
+    async def _execBackupTask(self, dirn):
+        '''
+        A task that backs up the cell to the target directory
+        '''
         await self.boss.promote('backup', self.auth.rootuser)
-        todo = s_common.todo(s_t_backup.backup, self.dirn, dirn)
-        await s_coro.spawn(todo)
+        slabs = s_lmdbslab.Slab.getSlabsInDir(self.dirn)
+        assert slabs
+
+        ctx = multiprocessing.get_context('spawn')
+
+        mypipe, child_pipe = ctx.Pipe()
+        paths = [str(slab.path) for slab in slabs]
+
+        def spawnproc():
+            proc = ctx.Process(target=self._backupProc, args=(child_pipe, self.dirn, dirn, paths))
+            proc.start()
+            hasdata = mypipe.poll(timeout=self.BACKUP_SPAWN_TIMEOUT)
+            if not hasdata:
+                raise s_exc.SynErr(mesg='backup subprocess stuck')
+            data = mypipe.recv()
+            assert data == 'ready'
+            return proc
+
+        proc = await s_coro.executor(spawnproc)
+
+        while True:
+            await s_lmdbslab.Slab.syncLoopOnce()
+            if not any(slab.dirty for slab in slabs):
+                break
+
+        try:
+            mypipe.send('proceed')
+
+            # This is technically pending the ioloop waiting for the backup process to acquire a bunch of
+            # transactions.  We're effectively locking out new write requests the brute force way.
+            hasdata = mypipe.poll(timeout=self.BACKUP_ACQUIRE_TIMEOUT)
+            if not hasdata:
+                raise s_exc.SynErr(mesg='backup subprocess stuck')
+
+            data = mypipe.recv()
+            assert data == 'captured'
+
+            def waitforproc():
+                proc.join()
+                if proc.exitcode:
+                    raise s_exc.SpawnExit(code=proc.exitcode)
```

Comment:
is it possible for us to capture stderr/stdout here?

Reply:
I think we'd rather have them go to whatever stdout/stderr is mapped in the parent process, no?

Reply:
I concur. s_coro.executor/run_in_executor does that for us.